planck-cljs/src/planck/io.cljs

```diff
@@ -552,6 +552,30 @@
   :args (s/cat :input any? :output any? :opts (s/* any?))
   :ret nil?)
 
+(defn tty?
+  "Returns true if x is a file descriptor associated with a terminal,
+  or x is either a Reader/Writer among *in*, *out*, or *err* which is
+  associated with a terminal.
+
+  Returns false if x is a file descriptor, *in*, *out*, or *err* and
+  not associated with a terminal.
+
+  Returns nil otherwise."
+  [x]
+  (when-let [fd (cond-> x
+                  (not (and (integer? x) (>= x 0)))
+                  {planck.core/*in*  0
+                   cljs.core/*out*   1
+                   planck.core/*err* 2})]
+    (js/PLANCK_ISATTY fd)))
+
+(s/fdef tty?
+  :args (s/cat :x (s/or :fd-num (s/and integer? (complement neg?))
+                        :reader #(implements? planck.core/IReader %)
+                        :writer #(implements? planck.core/IWriter %)))
+  :ret (s/or :tty? boolean?
+             :invalid nil?))
```

Comment:
My preference would be to coerce the return value by wrapping the call to `js/PLANCK_ISATTY` with a call to `boolean`. Generally, predicates return Boolean values. 

It seems that there is a strong desire from the community for this (some of the proposed new predicates in `clojure.core` that initially didn't return Boolean values, which were revised to do so. 

Also, in ClojureScript, there is a slight perf benefit as users of a predicate which returns a Boolean value: Owing to return type inference, a predicate returning a Boolean in an `if` test will result in more efficient, unchecked `if` code being generated. (See http://blog.fikesfarm.com/posts/2015-12-04-boolean-type-hints-in-clojurescript.html)